demisto_sdk/commands/split/jsonsplitter.py

```diff
@@ -124,4 +145,56 @@ def create_module(self):
 
             module_file_path = os.path.join(self.module_dir, file_name)
 
-        write_dict(module_file_path, data=self.module_json_data, indent=4)
+        write_dict(module_file_path, data=self.json_data, indent=JSON_INDENT_CONSTANT)
+
+    def get_auto_output_path(self) -> Tuple[Path, str, str]:
+        suffix_by_type = {
+            TypeList.TEXT: ".txt",
+            TypeList.HTML: ".html",
+            TypeList.CSS: ".css",
+            TypeList.MD: ".md",
+            TypeList.JSON: ".json",
+        }
+
+        suffix = suffix_by_type.get(self.json_data["type"], ".txt")
+
+        file_name = pascal_case(self.json_data["name"]) + ".json"
+        file_data_name = file_name[: -len(".json")] + "_data" + suffix
+
+        if self.autocreate_dir:
+            pack_name = get_pack_name(self.input)
+
+            if not pack_name:
+                return Path(self.input).parent, file_name, file_data_name
+
+            if not (lists_dir := Path(PACKS_DIR) / pack_name / LISTS_DIR).exists():
+                lists_dir.mkdir()
+
+            # create the specific list dir under the LISTS_DIR if it does not exist
+            # the dir name determine by the file name without the suffix
+            if not (list_name_dir := lists_dir / file_name[: -len(".json")]).exists():
+                list_name_dir.mkdir()
+
+            return list_name_dir, file_name, file_data_name
+
+        if not (output := Path(self.output)).is_dir():
+            output = output.parent
+
+        return output, file_name, file_data_name
+
+    def write_file_data(self, list_name_dir: Path, file_data_name: str):
+        if file_data_name.endswith("json"):
+            JsonFile.write_file(self.json_data["data"], (list_name_dir / file_data_name), indent=JSON_INDENT_CONSTANT)
+        else:
+            TextFile.write_file(self.json_data["data"], (list_name_dir / file_data_name))
+
+    def write_list(self, list_name_dir: Path, file_name: str):
+        self.json_data["data"] = "-"
```

Comment:
why do you put "-" in data?

Reply:
Similar to the implementation of unify in scripts and integrations, we set the data/script to "-"
but we also allow to leave it empty

Reply:
Please add a comment mentioning this